docs/tutorials/services/color-detection-scuttle.md

```diff
@@ -17,105 +17,383 @@ languages: [ "python" ]
+      return -1
+   for d in detections:
+      a = (d.x_max - d.x_min) * (d.y_max-d.y_min)
+      if a > largest_area:
+         a = largest_area
+         largest = d
+   centerX = largest.x_min + largest.x_max/2
+   if centerX < midpoint-midpoint/6:
+      return 0 # on the left
+   if centerX > midpoint+midpoint/6:
+      return 2 # on the right
+   else:
+      return 1  # basically centered
+```
+
+### Add the main function
+
+The `main` function:
+
+- defines variables for how the robot should move,
+- connects to the robot,
+- initializes the base, the camera, and the detector, and
+- repeatedly calls the `leftOrRight` function and turns the rover's base into the respective direction.
+
+Replace the `main` function with the following code:
+
+```python {class="line-numbers linkable-line-numbers"}
+async def main():
+   spinNum = 10         # when turning, spin the motor this much
+   straightNum = 300    # when going straight, spin motor this much
+   numCycles = 200      # run the loop X times
+   vel = 500            # go this fast when moving motor
+​
+   # Connect to robot client and set up components
+   robot = await connect()
+   base = Base.from_robot(robot, "base")
+   camera = Camera.from_robot(robot, "<camera-name>")
+
+   # Grab the vision service for the detector
+   my_detector = VisionClient.from_robot(robot, "my_color_detector")
+
+   # Main loop. Detect the ball, determine if it's on the left or right, and head that way.
+   # Repeat this for numCycles
+   for i in range(numCycles):
+      detections = await my_detector.get_detections_from_camera(camera)
+​
+      answer = leftOrRight(detections, frame.size[0]/2)
+      if answer == 0:
+         print("left")
+         await base.spin(spinNum, vel)  # CCW is positive
+         await base.move_straight(straightNum, vel)
+      if answer == 1:
+         print("center")
+         await base.move_straight(straightNum, vel)
+      if answer == 2:
+         print("right")
+         await base.spin(-spinNum, vel)
+      # If nothing is detected, nothing moves
+
+   await robot.close()​
+```
+
+For `<camera-name>`, insert the name of your configured physical camera.
+
+## Run the code
+
+Now, run the code:
+
+```sh {class="command-line" data-prompt="$"}
+python3 main.py
+```
+
+If everything works, your rover should detect and navigate towards red objects.
+Use something like a red sports ball or book cover as a target to follow to test your rover:
+
+<div class="aligncenter">
+{{<video webm_src="/tutorials/videos/scuttledemos_colordetection.webm" mp4_src="/tutorials/videos/scuttledemos_colordetection.mp4" poster="/tutorials/scuttlebot/scuttledemos_colordetection.jpg" alt="Detecting color with a Scuttle Robot">}}
+</div>
+
+## Next Steps
+
+Congratulations! If you're ready for more, try making your rover detect other colors.
+You could also write some code with a Viam SDK to [make your rover move in a square](/tutorials/get-started/try-viam-sdk/).
+
+{{< snippet "social.md" >}}
```

Comment:
nit: technically this also appears at the bottom, below full code. I'm OK with it though.